src/Microsoft.AspNet.WebUtilities/FormReader.cs

```diff
@@ -195,7 +195,24 @@ private void Buffer()
         /// </summary>
         /// <param name="stream">The HTTP form body to parse.</param>
         /// <returns>The collection containing the parsed HTTP form body.</returns>
-        public static async Task<IDictionary<string, StringValues>> ReadFormAsync(Stream stream, Encoding encoding, CancellationToken cancellationToken = new CancellationToken())
+        public static async Task<Dictionary<string, StringValues>> ReadFormAsync(Stream stream, Encoding encoding, CancellationToken cancellationToken = new CancellationToken())
+        {
+            var result = await ReadNullableFormAsync(stream, encoding, cancellationToken);
+
+            if (result == null)
+            {
+                return new Dictionary<string, StringValues>();
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Parses an HTTP form body - internal use, returns null if no entires.
+        /// </summary>
+        /// <param name="stream">The HTTP form body to parse.</param>
+        /// <returns>The collection containing the parsed HTTP form body, null if there are no entries.</returns>
+        public static async Task<Dictionary<string,StringValues>> ReadNullableFormAsync(Stream stream, Encoding encoding, CancellationToken cancellationToken = new CancellationToken())
```

Comment:
@benaadams - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormValueProviderFactory.cs#L27

We don't lift a finger unless it's really a formdata request.